src/Core/RulesPluginTrait.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Core\RulesPluginTrait.
+ */
+
+namespace Drupal\rules\Core;
+
+use Drupal\Component\Plugin\Exception\ContextException;
+
+/**
+ * Offers common methods for context plugin implementers.
+ */
+trait RulesPluginTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  private function getLabelValue() {
+    $definition = $this->getPluginDefinition();
```

Comment:
Due to naming confusion I preferred to get the label first and then make it accessible through the public function. The annotation sets the label, but the Plugin (as well as Plugin/Condition/Language::summary) kinda uses this summary function. I thought, that would be a more consistent way.

But feel free to rename it.